lib/ovirt/service.rb

```diff
@@ -145,8 +155,48 @@ def status(link)
       node.xpath('status/state').text
     end
 
+    # Checks if the API is available in the given candidate path. It does so sending a request without
+    # authentication. If the API is available there it will respond with the "WWW-Autenticate" header
+    # and with the "RESTAPI" or "ENGINE" realm.
+    def probe_api_path(uri, path)
+      uri = URI.join(uri, path)
+      request = RestClient::Resource.new(uri.to_s, :verify_ssl => OpenSSL::SSL::VERIFY_NONE)
+      begin
+        request.get
+      rescue RestClient::Exception => exception
+        response = exception.response
+        if response.code == 401
+          www_authenticate = response.headers[:www_authenticate]
+          if www_authenticate =~ /^Basic realm="?(RESTAPI|ENGINE)"?$/
+            return true
+          end
+        end
+      end
+      false
+    end
+
+    # Probes all the candidate paths of the API, and returns the first that success. If all probes
+    # fail, then the first candidate will be returned.
+    def find_api_path(uri)
+      CANDIDATE_API_PATHS.each do |path|
+        if probe_api_path(uri, path)
+          return path
+        end
+      end
+      CANDIDATE_API_PATHS[0]
+    end
+
+    # Returns the path of the API, probing it if needed.
+    def api_path
+      if @api_path.blank?
+        @api_path = find_api_path(base_uri)
+      end
```

Comment:
Actually, since this is the only setter for @api_path, you can just use the `||=` style:

```ruby
def api_path
  @api_path ||= find_api_path(base_uri)
end
```